src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlTransformRoutedEvent.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.Transformers;
+using XamlX.Ast;
+using XamlX.Emit;
+using XamlX.IL;
+using XamlX.Transform;
+using XamlX.TypeSystem;
+
+namespace Avalonia.Markup.Xaml.Loader.CompilerExtensions.Transformers;
+
+internal class AvaloniaXamlIlTransformRoutedEvent : IXamlAstTransformer
+{
+    public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode node)
+    {
+        if (node is XamlAstNamePropertyReference prop
+            && prop.TargetType is XamlAstClrTypeReference targetRef
+            && prop.DeclaringType is XamlAstClrTypeReference declaringRef)
+        {
+            var xkt = context.GetAvaloniaTypes();
+            var interactiveType = xkt.Interactivity.Interactive;
+            var routedEventType = xkt.Interactivity.RoutedEvent;
+            var AddHandlerT = xkt.Interactivity.AddHandlerT;
+
+            if (interactiveType.IsAssignableFrom(targetRef.Type))
+            {
+                var eventName = $"{prop.Name}Event";
+                if (declaringRef.Type.GetAllFields().FirstOrDefault(f => f.IsStatic && f.Name == eventName) is { } eventField)
+                {
+                    if (routedEventType.IsAssignableFrom(eventField.FieldType))
+                    {
+                        var instance = new XamlAstClrProperty(prop
+                            , prop.Name
+                            , targetRef.Type
+                            , null
+                            );
+                        instance.Setters.Add(new XamlDirectCallAddHandler(eventField,
+                            targetRef.Type,
+                            xkt.Interactivity.AddHandler,
+                            xkt.Interactivity.RoutedEventHandler
+                            )
+                        );
+                        if (eventField.FieldType.GenericArguments?.Count == 1)
+                        {
+                            var agrument = eventField.FieldType.GenericArguments[0];
+                            if (!agrument.Equals(xkt.Interactivity.RoutedEventArgs))
+                            {
+                                instance.Setters.Add(new XamlDirectCallAddHandler(eventField,
+                                    targetRef.Type,
+                                    xkt.Interactivity.AddHandlerT.MakeGenericMethod(new[] { agrument }),
+                                    context.Configuration.TypeSystem.FindType("System.EventHandler`1").MakeGenericType(agrument)
+                                    )
+                                );
+                            }
+                        }
+                        return instance;
+                    }
+                    else
+                    {
+                        //TODO: Throw Exception?
```

Comment:
I'm not sure whether to throw the exception or not

Reply:
If otherwise we have a runtime error: yes, exception.
If otherwise we have an unintended behavior, but app works: a new XAML warning.

Reply:
If I do nothing, there is "PropertyReferenceResolver" which throws `Unable to find suitable setter or adder for property`.  
I think it will generate a warning for now.